ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/PaywallViewModel.kt

```diff
@@ -110,8 +110,11 @@ internal class PaywallViewModelImpl(
     }
 
     fun updateOptions(options: PaywallOptions) {
-        if (this.options != options) {
-            this.options = options
+        val needsUpdateState = this.options.hashCode() != options.hashCode()
```

Comment:
If I'm not wrong, the regular equals check would work now that we have overridden the `hashCode()`. Did you happen to try this? 
```suggestion
        val needsUpdateState = this.options != options
```

Reply:
Ohh nice to know! I also assumed the same. Thanks for the investigation! 🤩